src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue

```diff
@@ -14,172 +14,276 @@
+                </router-link>
+                <router-link
+                  :to="editTreeUrl"
+                  event=""
+                  :title="trans('flow-builder.click-to-toggle-editing')"
+                  class="btn btn-outline-primary btn-sm"
+                  :class="{active: isEditable}"
+                  @click.native.prevent="handlePersistFlow(editTreeUrl)">
+                  {{ trans('flow-builder.edit-flow') }}
+                </router-link>
+              </div>
+
+              <slot name="extra-buttons" />
             </div>
 
-            <router-link
-              v-if="!ui.isEditableLocked"
-              :to="editOrViewTreeJsUrl"
-              event=""
-              :title="trans('flow-builder.click-to-toggle-editing')"
-              class="btn btn-secondary mr-2"
-              :class="{active: isEditable}"
-              @click.native.prevent="handlePersistFlow(editOrViewTreeJsUrl)">
-              {{ isEditable ? trans('flow-builder.view-flow') : trans('flow-builder.edit-flow') }}
-            </router-link>
-
-            <div
-              v-if="isEditable"
-              class="dropdown mr-2">
+            <div class="ml-auto mr-2">
               <button
+                v-if="hasSimulator"
                 type="button"
-                class="btn btn-secondary dropdown-toggle"
-                data-toggle="dropdown">
-                {{ trans('flow-builder.add-block') }}
+                class="btn btn-outline-primary btn-sm"
+                @click="setSimulatorActive(true)">
+                {{ trans('flow-builder.show-clipboard-simulator') }}
               </button>
 
-              <div class="dropdown-menu">
-                <template v-for="(classDetails, className) in rootBlockClassesToDisplay">
+              <div class="btn-group mr-2">
+                <slot name="right-grouped-buttons" />
+              </div>
+
+              <button
+                class="btn btn-outline-primary btn-sm"
+                :class="{active: isImporterVisible}"
+                @click="toggleImportExport">
+                {{ trans('flow-builder.import-export') }}
+              </button>
+
+              <!--TODO - do disable if no changes logic-->
+              <button
+                v-if="isEditable && isFeatureTreeSaveEnabled"
+                type="button"
+                class="btn btn-info btn-sm tree-save-tree ml-4"
+                :title="trans('flow-builder.save-changes-to-the-flow')"
+                :disabled="!!isTreeSaving"
+                @click="handlePersistFlow()">
+                {{ saveButtonText }}
+              </button>
+            </div>
+          </div>
+        </div>
+      </div>
+
+      <div v-if="isEditable" class="tree-workspace-panel-heading panel-heading w-100 bg-white d-flex justify-content-start pt-0 pb-0">
+        <div class="tree-workspace-panel-heading-contents">
+          <ul class="nav">
+            <li class="nav-item dropdown nav">
```

Comment:
Out of curiosity, is there a reason that we didn't scrap what was here in favour of: https://bootstrap-vue.org/docs/components/dropdown ?